core/trace/src/main/scala/org/typelevel/otel4s/trace/Tracer.scala

```diff
@@ -28,6 +28,11 @@ trait Tracer[F[_]] extends TracerMacro[F] {
     */
   def meta: Tracer.Meta[F]
 
+  /** Returns the span from the scope, falling back to a noop if none is
+    * available.
+    */
+  def currentSpan: F[Span[F]]
```

Comment:
This ship may have already sailed, but this is the signature that I think prevents using `Kleisli` (instead of `IOLocal`)
- https://github.com/typelevel/otel4s/issues/88

Reply:
I think that ship sailed with the `Resource[F, Span[F]]` that are already exposed via `TracerMacro`.

This is why Natchez's resource is a natural transformation: https://github.com/tpolecat/natchez/pull/526.  I do agree we need to decide on #88 before moving forward.

Reply:
Like, is this a new API, or is it natchez-with-lazy-macros?

Reply:
> this is the signature that I think prevents using `Kleisli` (instead of `IOLocal`)

So actually I'm not sure if this is true now. After getting very lost in https://github.com/tpolecat/natchez/pull/713 (and dragging poor @bpholt into my delusions 😅 ) I realized, isn't that PR doing exactly this? i.e. making it possible to access the `Span[F]` within `F`, without compromising a `Kleisli` implementation.

Reply:
I went down that same road for a while with Natchez and turned around.  I had a hard time with the inner `F` being a `Kleisli`, however much I wanted the outer `F` to be a `Kleisli`, and I didn't want to introduce a `G`.

Note that in our case, the local environment is not a `Span[F]`, but a (right now package-private) `TraceScope.Scope`.